src/api/source/mvt.js

```diff
@@ -68,70 +68,64 @@ export default class MVT extends Base {
         }
         const tile = new VectorTile(new Protobuf(arrayBuffer));
         const mvtLayer = tile.layers[Object.keys(tile.layers)[0]];
-
-        const { points, featureGeometries, properties } = this._decodeMVTLayer(mvtLayer, this._metadata, MVT_EXTENT);
-
-
+        const { geometries, properties } = this._decodeMVTLayer(mvtLayer, this._metadata, MVT_EXTENT);
         const rs = rsys.getRsysFromTile(x, y, z);
-        const dataframeGeometry = this._metadata.geomType == geometryTypes.POINT ? points : featureGeometries;
-        const dataframe = this._generateDataFrame(rs, dataframeGeometry, properties, mvtLayer.length, this._metadata.geomType);
+        const dataframe = this._generateDataFrame(rs, geometries, properties, mvtLayer.length, this._metadata.geomType);
+
         return dataframe;
     }
 
 
     _decodeMVTLayer(mvtLayer, metadata, mvt_extent) {
-        let points;
-        if (metadata.geomType == geometryTypes.POINT) {
-            points = new Float32Array(mvtLayer.length * 2);
+        switch (metadata.geomType) {
+            case geometryTypes.POINT:
+                return this._decode(mvtLayer, metadata, mvt_extent);
+            case geometryTypes.LINE:
+                return this._decode(mvtLayer, metadata, mvt_extent, decodeLines);
+            case geometryTypes.POLYGON:
+                return this._decode(mvtLayer, metadata, mvt_extent, decodePolygons);
+            default:
+                throw new Error('MVT: invalid geometry type');
         }
-        let featureGeometries = [];
-        const decodedProperties = {};
-        const decodingPropertyNames = Object.keys(metadata.properties).
-            filter(propertyName => metadata.properties[propertyName].type != 'geometry').
-            map(propertyName => metadata.properties[propertyName].sourceName || propertyName);
-        decodingPropertyNames.forEach(propertyName => {
-            decodedProperties[propertyName] = new Float32Array(mvtLayer.length + RTT_WIDTH);
-        });
+    }
+
+    _decode(mvtLayer, metadata, mvt_extent, decodeFn) {
+        const { properties, propertyNames } = this._getPropertyNames(metadata, mvtLayer.length);
+        const geometries = decodeFn ? [] : new Float32Array(mvtLayer.length * 2);
```

Comment:
The only thing here, instead of initializing the geometry by checking if there is a `decodeFn`, I would pass it as a parameter:

```js
    _decodeMVTLayer(mvtLayer, metadata, mvt_extent) {
        let geometries;
        
        switch (metadata.geomType) {
            case geometryTypes.POINT:
                geometries = new Float32Array(mvtLayer.length * 2);
                return this._decode(mvtLayer, metadata, geometries, mvt_extent);
            case geometryTypes.LINE:
                geometries = [];
                return this._decode(mvtLayer, metadata, geometries, mvt_extent, decodeLines);
            case geometryTypes.POLYGON:
                geometries = [];
                return this._decode(mvtLayer, metadata, geometries, mvt_extent, decodePolygons);
            default:
                throw new Error('MVT: invalid geometry type');
        }
    }
```